src/Adapter/Product/Repository/TagRepository.php

```diff
@@ -110,4 +131,33 @@ public function deleteTagsByLanguage(ProductId $productId, LanguageId $languageI
             ));
         }
     }
+
+    /**
+     * @param ProductId $productId
+     *
+     * @return array Localized tags for a product
+     */
+    public function getProductTags(ProductId $productId): array
```

Comment:
getProductTags or getLocalizedTags ?

Reply:
or getLocalizedProductTags ? (I think this is better according the it's used after)